common/src/main/java/com/wynntils/handlers/inventory/InventoryHandler.java

```diff
@@ -0,0 +1,449 @@
+                new ItemStack(Items.BARRIER, 31),
+                Int2ObjectMaps.emptyMap()));
+    }
+
+    private void forceSyncLater(AbstractContainerMenu menu) {
+        Managers.TickScheduler.scheduleNextTick(() -> forceSync(menu));
+    }
+
+    private void expect(PendingInteraction interaction) {
+        pending.add(new PendingInteractionEntry(interaction));
+    }
+
+    private static boolean areItemsSimilar(ItemStack a, ItemStack b) {
+        if (a.isEmpty()) {
+            return b.isEmpty();
+        } else {
+            return !b.isEmpty()
+                    && a.getHoverName().getString().equals(b.getHoverName().getString());
+        }
+    }
+
+    private static boolean areStacksSimilar(ItemStack a, ItemStack b) {
+        return areItemsSimilar(a, b) && a.getCount() == b.getCount();
+    }
+
+    private final class RunningInteraction {
+        private final AbstractContainerMenu menu;
+        private final boolean single;
+        private final ItemStack heldStack;
+        private final IntList slots = new IntArrayList();
+
+        private RunningInteraction(AbstractContainerMenu menu, boolean single, ItemStack heldStack) {
+            this.menu = menu;
+            this.single = single;
+            this.heldStack = heldStack;
+        }
+
+        private boolean handle(ContainerClickEvent event) {
+            if (event.getClickType() != ClickType.QUICK_CRAFT
+                    || event.getContainerMenu().containerId != menu.containerId) {
+                running = null;
+                return false;
+            }
+
+            int mask = event.getMouseButton();
+            if (((mask & 0x4) == 0) == single) {
+                running = null;
+                return false;
+            }
+
+            switch (event.getMouseButton() & 0x3) {
+                case 1 -> slots.add(event.getSlotNum());
+                case 2 -> {
+                    forceSyncLater(menu);
+                    expect(new PendingInteraction.Spread(menu, slots, heldStack, single));
+                    running = null;
+                }
+                default -> {
+                    running = null;
+                    return false;
+                }
+            }
+            return true;
+        }
+    }
+
+    private interface PendingInteraction {
+        boolean onSlotUpdate(int containerId, int slotNum, ItemStack newStack);
+
+        default boolean onContentUpdate(int containerId, List<ItemStack> newInv) {
+            return true;
+        }
+
+        abstract class CaptureHeld implements PendingInteraction {
+            @Override
+            public boolean onSlotUpdate(int containerId, int slotNum, ItemStack newStack) {
+                if (containerId == -1) {
+                    tryDispatch(newStack);
+                    return false;
+                }
+                return true;
+            }
+
+            protected abstract void tryDispatch(ItemStack newHeldStack);
```

Comment:
The name `tryDispatch` throw me off a bit when trying to understand this code. I recommend `sendEvent` instead. The "try" part is not needed; you can always assume that any method will only perform its core functionality if the correct conditions hold. And we use the terminology "send", not "dispatch", of events.

Reply:
Sure, that's okay.